onnxruntime/core/optimizer/qdq_transformer/qdq_s8_to_u8.h

```diff
@@ -16,11 +16,15 @@ namespace onnxruntime {
 class QDQS8ToU8Transformer : public GraphTransformer {
  public:
   QDQS8ToU8Transformer(bool weights_to_u8, const InlinedHashSet<std::string_view>& compatible_execution_providers = {}) noexcept
-      : GraphTransformer("QDQS8ToU8Transformer", compatible_execution_providers), weights_to_u8_(weights_to_u8) {}
+      : GraphTransformer("QDQS8ToU8Transformer", compatible_execution_providers), weights_to_u8_(weights_to_u8) {
+          exclude_next = {"MatMul", "Conv", "Gemm", "Gather"};
```

Comment:
It is good to also convert s8 to u8 for Gather.

Reply:
This test is not passing anymore if I remove `Gather` from the list: https://github.com/microsoft/onnxruntime/blob/81b128b5e97284f576ed8aaf18404980cb45fe34/onnxruntime/test/optimizer/qdq_transformer_test.cc#L738-L766

with the errors:
```
1: Expected equality of these values:
1:   op_to_count["QuantizeLinear"]
1:     Which is: 1
1:   0
1: Google Test trace:
1: /home/fxmarty/hf_internship/onnxruntime/onnxruntime/test/common/tensor_op_test_utils.cc:14: ORT test random seed: 2345
1: /home/fxmarty/hf_internship/onnxruntime/onnxruntime/test/optimizer/qdq_transformer_test.cc:799: Failure
1: Expected equality of these values:
1:   op_to_count["DequantizeLinear"]
1:     Which is: 1
1:   0
```
Any idea why?